protocol/txntype.go

```diff
@@ -47,3 +47,31 @@ const (
 	// UnknownTx signals an error
 	UnknownTx TxType = "unknown"
 )
+
+// TxTypeToByte converts a TxType to byte encoding
+func TxTypeToByte(t TxType) byte {
+	switch t {
+	case PaymentTx:
+		return 0
+	case KeyRegistrationTx:
+		return 1
+	case AssetConfigTx:
+		return 2
+	case AssetTransferTx:
+		return 3
+	case AssetFreezeTx:
+		return 4
+	case ApplicationCallTx:
+		return 5
+	case CompactCertTx:
+		return 6
+	default:
+		return 7
+	}
+}
+
+// ByteToTxType converts a byte encoding to TxType
+func ByteToTxType(b byte) TxType {
+	txTypes := []TxType{PaymentTx, KeyRegistrationTx, AssetConfigTx, AssetTransferTx, AssetFreezeTx, ApplicationCallTx, CompactCertTx, UnknownTx}
+	return txTypes[b]
+}
```

Comment:
These would better be in txsync and not in protocol. 
consider defining an enum for these instead of using numeric values.